module/icp/core/kcf_sched.c

```diff
@@ -1731,20 +1731,22 @@ kcf_last_req(void *last_req_arg, int status)
 		kcf_mac_ops_params_t *mops = &(params->rp_u.mac_params);
 
 		ct = (crypto_dual_data_t *)mops->mo_data;
+		ct->dd_offset1 = last_req->kr_saveoffset;
+		ct->dd_len1 = last_req->kr_savelen;
 		break;
 	}
 	case KCF_OG_DECRYPT: {
 		kcf_decrypt_ops_params_t *dcrops =
 		    &(params->rp_u.decrypt_params);
 
 		ct = (crypto_dual_data_t *)dcrops->dop_ciphertext;
+		ct->dd_offset1 = last_req->kr_saveoffset;
+		ct->dd_len1 = last_req->kr_savelen;
 		break;
 	}
 	default:
 		break;
```

Comment:
So this should never happen, but if it were to happen wasn't the most graceful way to deal with it.  Simply return in the `default` case, ASSERT, or attempt to carry on as is done by this patch.